pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -577,9 +577,25 @@ public static Properties parseURL(String url, Properties defaults) {
       urlProps.setProperty("PGPORT", ports.toString());
       urlProps.setProperty("PGHOST", hosts.toString());
     } else {
-      urlProps.setProperty("PGPORT", "/*$mvn.project.property.template.default.pg.port$*/");
-      urlProps.setProperty("PGHOST", "localhost");
-      urlProps.setProperty("PGDBNAME", URLDecoder.decode(l_urlServer));
+
+      if (defaults != null && !defaults.containsKey("PGPORT")) {
+        urlProps.setProperty("PGPORT", "/*$mvn.project.property.template.default.pg.port$*/");
+      } else {
+        urlProps.setProperty("PGPORT", "/*$mvn.project.property.template.default.pg.port$*/");
```

Comment:
The above looks like an unconditional update of `urlProps.setProperty("PGPORT"`. Is `if` required at all in such case? Was something other than `urlProps` expected here?